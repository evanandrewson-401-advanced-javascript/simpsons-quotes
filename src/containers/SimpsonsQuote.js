import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Quote from '../components/quote/Quote';
import { connect } from 'react-redux';
import { getLoading, getCharacterImage, getCharacterName, getQuote } from '../selectors/quoteSelectors';
import { setQuotePromise } from '../actions/quoteDataActions';
import Load from '../components/quote/Load';

class SimpsonsQuote extends Component {
  componentDidMount() {
    this.props.fetchQuote();
  }

  render() {
    if(this.props.loading) return <h1>Loading...</h1>;
    return (
      <>
        <Quote quote={this.props.quote} characterName={this.props.characterName} characterImage={this.props.characterImage}  />
        <Load fetchQuote={this.props.fetchQuote} />
      </>
    );
  }

  static propTypes = {
    quote: PropTypes.string.isRequired,
    characterName: PropTypes.string.isRequired,
    characterImage: PropTypes.string.isRequired,
    fetchQuote: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired
  }
}

const mapStateToProps = state => ({
  quote: getQuote(state),
  characterName: getCharacterName(state),
  characterImage: getCharacterImage(state),
  loading: getLoading(state)
});

const mapDispatchToProps = dispatch => ({
  fetchQuote() {
    dispatch(setQuotePromise());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SimpsonsQuote);
