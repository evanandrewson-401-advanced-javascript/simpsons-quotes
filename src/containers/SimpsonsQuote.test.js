import React from 'react';
import { shallow } from 'enzyme';
import SimpsonsQuote from './SimpsonsQuote';
import store from '../store';

describe('SimpsonsQuote component', () => {
  it('renders SimpsonsQuote', () => {
    const wrapper = shallow(<SimpsonsQuote store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
