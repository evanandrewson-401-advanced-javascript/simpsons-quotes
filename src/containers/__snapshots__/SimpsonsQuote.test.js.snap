// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpsonsQuote component renders SimpsonsQuote 1`] = `
<ContextProvider
  value={null}
>
  <SimpsonsQuote
    characterImage=""
    characterName=""
    fetchQuote={[Function]}
    loading={false}
    quote=""
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
